tests/test_search.py

```diff
@@ -1139,3 +1142,47 @@ def test_missing_fields(postgresql, mongo, backend, rc, tmp_path):
     data = resp.json()
     assert resp.status_code == 200, data
     assert take(data) == {'code': 'lt'}
+
+
+def test_base_select(rc, postgresql, request):
+    context = bootstrap_manifest(rc, '''
+    d | r | b | m | property   | type    | ref
+    datasets/gov/example/base  |         |
+                               |         |
+      |   |   | Location       |         |
+      |   |   |   | id         | integer |
+      |   |   |   | name       | string  |
```

Comment:
'type' can be selected only when it is in City's model property list. If 'type' was added to structure as City's property (with Inherit type) or if we called select(_base.type), it would work. I'm not really sure how to solve this, because if we want to select just the 'type', then during model load we should take base model properties and copy them to City's model properties with Inherit type, the problem with that is that during load base model could be not loaded yet, so we can't access it's property list. And creating new properties during link command doesn't really sound right.

Reply:
If `type` is defined on `Location`, then it should not be defined or added to `City` in any way.

When a query like `City?select(type)` is given, then we should check if `type` is part of `City` properties, if not, then we should check if `type` if part of `City` base properties and if yes, then we should add a join with `Location`.

But this can be postponed and moved into a separate task, unless it is pretty trivial to implement.